### PR TITLE
correct help text for Block glyph coordinates

### LINKED ISSUE #14173: [BUG] Help text for Block x and y coordinates  is incorrect
### Software versions

N/A

### Browser name and version

_No response_

### Jupyter notebook / Jupyter Lab version

_No response_

### Expected behavior

The [documentation](https://docs.bokeh.org/en/latest/docs/reference/models/glyphs/block.html) states that the Block "Render rectangular regions, given a **corner** coordinate, width, and height.".

Further down, 'x' and `y` are described as "The x-coordinates of the **centers** of the blocks."

Corner is correct. I've made a PR to change it.

### Observed behavior

Corner, not center.

### Example code

```Python
N/A
```

### Stack traceback or browser console output

_No response_

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by henricryden**
- [x] issues: fixes #14173

### CODE REVIEW

**Comment by mosc9575** [comment]
@henricryden Thanks for opening this PR. With your help we can improve the documentation of the project.

Two short comments:

1. In my opinion `a corner` is not specific enough. It is the `lower-left` corner. I think it would make sense to be more precise. There are two places it should be corrected.

https://github.com/bokeh/bokeh/blob/c829640ac4e244e55153c3f5d5f70d9b9bda320d/docs/bokeh/source/docs/user_guide/basic/areas.rst?plain=1#L31

https://github.com/bokeh/bokeh/blob/c829640ac4e244e55153c3f5d5f70d9b9bda320d/src/bokeh/models/glyphs.py#L414

2. There is a syntax error in the documentation for the `figure.block()` example. This could be fixed in the PR, too.

https://github.com/bokeh/bokeh/blob/c829640ac4e244e55153c3f5d5f70d9b9bda320d/src/bokeh/plotting/glyph_api.py#L161

Do you want to make this changes?

If you need help please feel free to ask.

**Comment by henricryden** [comment]
Thanks, I've made the changes you mentioned. While this deals with the issue, I want to ask you if you think the anchor point of the block glyph should be user configurable? Lower-left is a bit arbitrary, but that's just my opinion of course.

File: src/bokeh/models/glyphs.py [hunk 1]
```diff
@@ -424,11 +424,11 @@ def __init__(self, *args, **kwargs) -> None:
     _args = ('x', 'y', 'width', 'height')
 
     x = NumberSpec(default=field("x"), help="""
-    The x-coordinates of the centers of the blocks.
+    The x-coordinates of each block's lower-left corner
```
Comment (mosc9575) [inline src/bokeh/models/glyphs.py]:
```suggestion
    The x-coordinates of each block's lower-left corner.
```

Missing dot.

File: src/bokeh/models/glyphs.py [hunk 2]
```diff
@@ -424,11 +424,11 @@ def __init__(self, *args, **kwargs) -> None:
...
     """)
 
     y = NumberSpec(default=field("y"), help="""
-    The y-coordinates of the centers of the blocks.
+    The y-coordinates of each block's lower-left corner
```
Comment (mosc9575) [inline src/bokeh/models/glyphs.py]:
```suggestion
    The y-coordinates of each block's lower-left corner.
```

**Comment by mosc9575** [comment]
> I want to ask you if you think the anchor point of the block glyph should be user configurable?

This idea could become a feature request but I don't think this is necessary, because bokeh offers three different options to draw rectangular areas

* [quad](https://docs.bokeh.org/en/latest/docs/user_guide/basic/areas.html#quads) given by lower-left and upper-right corner
* [block](https://docs.bokeh.org/en/latest/docs/user_guide/basic/areas.html#blocks) given by lower-left and width and height
* [rect](https://docs.bokeh.org/en/latest/docs/user_guide/basic/areas.html#rotatable) given by the center, width, height corner and an angle

In my opinion this should be sufficient for the users. But I am open for other opinions.

**Comment by bryevdv** [comment]
AFAIK widths and heights can be negative, in which cases the corner would not be "lower left" in a practical sense. Though I'm not in front of a computer to check for certain.

**Comment by mosc9575** [comment]
@bryevdv Here is the original example from [blocks](https://docs.bokeh.org/en/latest/docs/user_guide/basic/areas.html#blocks) and an adapted version.

##### Example with **positive** width and height

```python
from bokeh.plotting import figure, show,

p = figure(width=400, height=400)
p.block(x=[1, 2, 3], y=[1, 2, 3], width=[0.2, 0.5, 0.1], height=1.5)

show(p)
```

grafik

##### Example with **negative** width and height

```python
from bokeh.plotting import figure, show

p = figure(width=400, height=400)
p.block(x=[1, 2, 3], y=[1, 2, 3], width=[-0.2, -0.5, -0.1], height=[-1.5]*3)

show(p)
```

|output|adjusted output with pan tool|
|--|--|
|grafik|grafik|

As you can see, the given corner is always the lower-left one, but the negative width and height has an effect on the x- and y-range of the figure.

#### Comment

The example below using a negative value for the height and not a list of negative values results in an error message.

```python
from bokeh.plotting import figure, show

p = figure(width=400, height=400)
p.block(x=[1, 2, 3], y=[1, 2, 3], width=[-0.2, -0.5, -0.1], height=-1.5)

show(p)
```

```ValueError: Distances must be positive!```

<details><summary>Traceback</summary>
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[10], [line 5](vscode-notebook-cell:?execution_count=10&line=5)
      [2](vscode-notebook-cell:?execution_count=10&line=2) output_notebook()
      [4](vscode-notebook-cell:?execution_count=10&line=4) p = figure(width=400, height=400)
----> [5](vscode-notebook-cell:?execution_count=10&line=5) p.block(x=[1, 2, 3], y=[1, 2, 3], width=[-0.2, -0.5, -0.1], height=-1.5)
      [7](vscode-notebook-cell:?execution_count=10&line=7) show(p)

File c:\Users\mosc\envs\fbpinn\Lib\site-packages\bokeh\plotting\_decorators.py:89, in glyph_method.<locals>.decorator.<locals>.wrapped(self, *args, **kwargs)
     [87](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/plotting/_decorators.py:87) if self.coordinates is not None:
     [88](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/plotting/_decorators.py:88)     kwargs.setdefault("coordinates", self.coordinates)
---> [89](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/plotting/_decorators.py:89) return create_renderer(glyphclass, self.plot, **kwargs)

File c:\Users\mosc\envs\fbpinn\Lib\site-packages\bokeh\plotting\_renderer.py:113, in create_renderer(glyphclass, plot, **kwargs)
    [110](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/plotting/_renderer.py:110) # handle the mute glyph, we always set one
    [111](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/plotting/_renderer.py:111) muted_visuals = pop_visuals(glyphclass, kwargs, prefix='muted_', defaults=glyph_visuals, override_defaults={'alpha':0.2})
--> [113](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/plotting/_renderer.py:113) glyph = make_glyph(glyphclass, kwargs, glyph_visuals)
    [114](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/plotting/_renderer.py:114) nonselection_glyph = make_glyph(glyphclass, kwargs, nonselection_visuals)
    [115](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/plotting/_renderer.py:115) selection_glyph = make_glyph(glyphclass, kwargs, selection_visuals)

File c:\Users\mosc\envs\fbpinn\Lib\site-packages\bokeh\plotting\_renderer.py:142, in make_glyph(glyphclass, kws, extra)
    [140](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/plotting/_renderer.py:140) kws = kws.copy()
    [141](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/plotting/_renderer.py:141) kws.update(extra)
--> [142](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/plotting/_renderer.py:142) return glyphclass(**kws)

File c:\Users\mosc\envs\fbpinn\Lib\site-packages\bokeh\models\glyphs.py:420, in Block.__init__(self, *args, **kwargs)
    [419](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/models/glyphs.py:419) def __init__(self, *args, **kwargs) -> None:
--> [420](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/models/glyphs.py:420)     super().__init__(*args, **kwargs)

File c:\Users\mosc\envs\fbpinn\Lib\site-packages\bokeh\models\glyphs.py:216, in LRTBGlyph.__init__(self, *args, **kwargs)
    [215](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/models/glyphs.py:215) def __init__(self, *args, **kwargs) -> None:
--> [216](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/models/glyphs.py:216)     super().__init__(*args, **kwargs)

File c:\Users\mosc\envs\fbpinn\Lib\site-packages\bokeh\models\glyph.py:117, in LineGlyph.__init__(self, *args, **kwargs)
    [116](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/models/glyph.py:116) def __init__(self, *args, **kwargs) -> None:
--> [117](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/models/glyph.py:117)     super().__init__(*args, **kwargs)

File c:\Users\mosc\envs\fbpinn\Lib\site-packages\bokeh\models\glyph.py:127, in FillGlyph.__init__(self, *args, **kwargs)
    [126](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/models/glyph.py:126) def __init__(self, *args, **kwargs) -> None:
--> [127](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/models/glyph.py:127)     super().__init__(*args, **kwargs)

File c:\Users\mosc\envs\fbpinn\Lib\site-packages\bokeh\models\glyph.py:147, in HatchGlyph.__init__(self, *args, **kwargs)
    [146](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/models/glyph.py:146) def __init__(self, *args, **kwargs) -> None:
--> [147](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/models/glyph.py:147)     super().__init__(*args, **kwargs)

File c:\Users\mosc\envs\fbpinn\Lib\site-packages\bokeh\models\glyph.py:63, in Glyph.__init__(self, *args, **kwargs)
     [62](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/models/glyph.py:62) def __init__(self, *args, **kwargs) -> None:
---> [63](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/models/glyph.py:63)     super().__init__(*args, **kwargs)

File c:\Users\mosc\envs\fbpinn\Lib\site-packages\bokeh\model\model.py:119, in Model.__init__(self, *args, **kwargs)
    [116](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/model/model.py:116) if "id" in kwargs:
    [117](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/model/model.py:117)     raise ValueError("initializing 'id' is not allowed")
--> [119](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/model/model.py:119) super().__init__(**kwargs)
    [120](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/model/model.py:120) default_theme.apply_to_model(self)

File c:\Users\mosc\envs\fbpinn\Lib\site-packages\bokeh\core\has_props.py:304, in HasProps.__init__(self, **properties)
    [302](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/has_props.py:302)     if value is Undefined or value is Intrinsic:
    [303](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/has_props.py:303)         continue
--> [304](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/has_props.py:304)     setattr(self, name, value)
    [306](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/has_props.py:306) initialized = set(properties.keys())
    [307](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/has_props.py:307) for name in self.properties(_with_props=True): # avoid set[] for deterministic behavior

File c:\Users\mosc\envs\fbpinn\Lib\site-packages\bokeh\core\has_props.py:336, in HasProps.__setattr__(self, name, value)
    [334](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/has_props.py:334) properties = self.properties(_with_props=True)
    [335](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/has_props.py:335) if name in properties:
--> [336](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/has_props.py:336)     return super().__setattr__(name, value)
    [338](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/has_props.py:338) descriptor = getattr(self.__class__, name, None)
    [339](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/has_props.py:339) if isinstance(descriptor, property): # Python property

File c:\Users\mosc\envs\fbpinn\Lib\site-packages\bokeh\core\property\descriptors.py:875, in UnitsSpecPropertyDescriptor.__set__(self, obj, value, setter)
    [840](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/descriptors.py:840) """ Implement the setter for the Python `descriptor protocol`_.
    [841](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/descriptors.py:841) 
    [842](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/descriptors.py:842) This method first separately extracts and removes any ``units`` field
   (...)
    [872](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/descriptors.py:872) 
    [873](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/descriptors.py:873) """
    [874](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/descriptors.py:874) value = self._extract_units(obj, value)
--> [875](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/descriptors.py:875) super().__set__(obj, value, setter=setter)

File c:\Users\mosc\envs\fbpinn\Lib\site-packages\bokeh\core\property\descriptors.py:330, in PropertyDescriptor.__set__(self, obj, value, setter)
    [327](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/descriptors.py:327)     class_name = obj.__class__.__name__
    [328](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/descriptors.py:328)     raise RuntimeError(f"{class_name}.{self.name} is a readonly property")
--> [330](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/descriptors.py:330) value = self.property.prepare_value(obj, self.name, value)
    [331](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/descriptors.py:331) old = self._get(obj)
    [332](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/descriptors.py:332) self._set(obj, old, value, setter=setter)

File c:\Users\mosc\envs\fbpinn\Lib\site-packages\bokeh\core\property\dataspec.py:493, in DistanceSpec.prepare_value(self, cls, name, value)
    [491](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/dataspec.py:491) try:
    [492](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/dataspec.py:492)     if value < 0:
--> [493](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/dataspec.py:493)         raise ValueError("Distances must be positive!")
    [494](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/dataspec.py:494) except TypeError:
    [495](file:///C:/Users/mosc/envs/fbpinn/Lib/site-packages/bokeh/core/property/dataspec.py:495)     pass

ValueError: Distances must be positive!

</details>

**Comment by bryevdv** [comment]
I see. Going back to the original PR for `Block` I see there was some discussion of the possibility of accepting signed distances, but this evidently this was not implemented at the time. I'd regard the second example above as undefined behavior at this point, so I support the use of "lower left" in the docs for now (until/unless support for anchors or signed distances is added).

**Comment by henricryden** [comment]
It seems to be this minmax call which handles negative width/height:

https://github.com/bokeh/bokeh/blob/0e7e63af931746e6e9c213f7f993366b986908cd/bokehjs/src/lib/models/glyphs/block.ts#L19-L29

Since `width` is a `DistanceSpec` [according to the documentation](https://docs.bokeh.org/en/latest/docs/reference/models/glyphs/block.html#bokeh.models.Block.width), my interpretation was that negative values aren't supported.
https://github.com/bokeh/bokeh/blob/0e7e63af931746e6e9c213f7f993366b986908cd/src/bokeh/core/property/dataspec.py#L481-L497

Regarding the different glyphs Bokeh offers, I have to admit that I found it a bit confusing.  I would prefer only one glyph for rectangles, with optional `angle`for rotations, as well as different anchor points. After all, they are all just rectangles 🙂 This is just my opinion and I understand that there are multiple ways you can approach this, and I'm sure there are reasons why there are currently three separate glyphs.

**Comment by bryevdv** [comment]
@henricryden That first codepath is shared by several glyphs. My speculation is that it was added to support signed distances in some other glyph (e.g. negative "top" for vbars), but that more work would be needed to add support for signed distances specifically to `Block`. FWIW I would be in favor or adding signed distance support to `Block` but that's something that should be hashed out in a new Issue or Discussion, in case anyone wants to start one.

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
Thanks for the PR @henricryden

**Pull request merged**